src/index.ts

```diff
@@ -46,9 +90,26 @@ export function setCookie(
     if (typeof cookies === 'string') cookies = [cookies]
     if (typeof cookies === 'number') cookies = []
 
-    cookies.push(cookie.serialize(name, value, options))
-
-    ctx.res.setHeader('Set-Cookie', cookies)
+    const parsedCookies = setCookieParser.parse(cookies)
+
+    let cookiesToSet: string[] = []
+    parsedCookies.forEach((parsedCookie: Cookie) => {
```

Comment:
Could we replace this with a more functional code? I was thinking of filtering the cookies instead of pushing and pooling from the list.

Reply:
This was a micro optimisation. With filter I need to filter the array first and then serialise all filtered values into the string. So the complexity was O(2n) and this algorithm require only O(n) although it has double memory consumption.

But if you insist I can rewrite it to `parsedCookies.filter().map()` instead

Reply:
@koprivajakub if you don't mind I would rather have it implemented in `filter` and `map`.